enigma-js/test/integrationTests/template.10_execute_voting.js

```diff
@@ -0,0 +1,259 @@
+    const epochSize = parseInt(await enigma.enigmaContract.methods.getEpochSize().call());
+    const epochRemains = (firstBlock + epochSize) - currentBlock;
+    for (let i = 0; i < epochRemains; i++) {
+      await sampleContract.methods.incrementCounter().send({from: accounts[8]});
+    }
+    // Wait for 2s for the Ppal node to pick up the new epoch
+    await sleep(3000);
+
+    const pollId = (await votingETHContract.methods.getPolls().call()).length - 1;
+    let taskFn = 'cast_vote(uint256,bytes32,uint256)';
+    let taskArgs = [
+      [pollId, 'uint256'],
+      [addr2, 'bytes32'],
+      [0, 'uint256'],
+    ];
+    let taskGasLimit = 1000000;
+    let taskGasPx = utils.toGrains(1);
+    task2 = await new Promise((resolve, reject) => {
+      enigma.computeTask(taskFn, taskArgs, taskGasLimit, taskGasPx, accounts[0], votingAddr)
+        .on(eeConstants.SEND_TASK_INPUT_RESULT, (result) => resolve(result))
+        .on(eeConstants.ERROR, (error) => reject(error));
+    });
+  }, 10000);
+
+  it('should get the pending task', async () => {
+    task2 = await enigma.getTaskRecordStatus(task2);
+    expect(task2.ethStatus).toEqual(1);
+  });
+
+  it('should get the confirmed task success', async () => {
+    do {
+      await sleep(1000);
+      task2 = await enigma.getTaskRecordStatus(task2);
+      process.stdout.write('Waiting. Current Task Status is '+task2.ethStatus+'\r');
+    } while (task2.ethStatus !== 2);
+    expect(task2.ethStatus).toEqual(2);
+    process.stdout.write('Completed. Final Task Status is '+task2.ethStatus+'\n');
+  }, 10000);
+
+  let task3;
+  it('should fail to execute compute task to tally poll when poll has not expired', async () => {
+    const pollId = (await votingETHContract.methods.getPolls().call()).length - 1;
+    let taskFn = 'tally_poll(uint256)';
+    let taskArgs = [
+      [pollId, 'uint256'],
+    ];
+    let taskGasLimit = 1000000;
+    let taskGasPx = utils.toGrains(1);
+    task3 = await new Promise((resolve, reject) => {
+      enigma.computeTask(taskFn, taskArgs, taskGasLimit, taskGasPx, accounts[0], votingAddr)
+        .on(eeConstants.SEND_TASK_INPUT_RESULT, (result) => resolve(result))
+        .on(eeConstants.ERROR, (error) => reject(error));
+    });
+  });
+
+  it('should get the pending task', async () => {
+    task3 = await enigma.getTaskRecordStatus(task3);
+    expect(task3.ethStatus).toEqual(1);
+  });
+
+  it('should get the confirmed task success', async () => {
+    do {
+      await sleep(1000);
+      task3 = await enigma.getTaskRecordStatus(task3);
+      process.stdout.write('Waiting. Current Task Status is '+task3.ethStatus+'\r');
+    } while (task3.ethStatus !== 4);
+    expect(task3.ethStatus).toEqual(4);
+    process.stdout.write('Completed. Final Task Status is '+task3.ethStatus+'\n');
+  }, 10000);
+
+  // it('checks poll is still pending on ETH', async () => {
```

Comment:
Any reason why you want to keep all these lines commented out here, vs removing them altogether?

Reply:
uncommented in this latest commit!